### PR TITLE
[Enhance] Refactor `imshow_det_rbboxes`

### PR DESCRIPTION
**Pull request by liuyanyi**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Follow mmdet https://github.com/open-mmlab/mmdetection/pull/6716
#121 
DOTA palette follow [BboxToolkit](https://github.com/jbwang1997/BboxToolkit/blob/master/tools/vis_configs/dota1_0/colors.txt)

before:
image
after:
image

## Modification
Use matplotlib instead of cv2.
Support visualize boxes of diff classes as diff colors.
Add palette for DOTA dataset
Remove `output` in `image_demo.py`

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by zytx121** [comment]
We could add `mmrotate.core.visualization` in `mmrotate/docs/zh_cn/api.rst` and `mmrotate/docs/en/api.rst`, so that we could generate the docs about  visualization API. (forgot before...)

**Comment by zytx121** [comment]
Why does the color in `MMRotate_Tutorial.ipynb` seems didn't change?

File: mmrotate/core/visualization/image.py [hunk 1]
```diff
@@ -1,107 +1,244 @@
...
-            closed = cv2.getWindowProperty(win_name, cv2.WND_PROP_VISIBLE) < 1
-            # if user closed window or if some key pressed
-            if closed or ret != -1:
-                break
-    else:
-        ret = cv2.waitKey(wait_time)
+
+def draw_bboxes(ax, bboxes, color='g', alpha=0.8, thickness=2):
+    """Draw bounding boxes on the axes.
+
```
Comment (jbwang1997) [inline mmrotate/core/visualization/image.py:42]: Can change the name to `draw_rbboxes` to make differences with mmdet.

**Comment by jbwang1997** [comment]
[huge_image_demo.py](https://github.com/open-mmlab/mmrotate/blob/main/demo/huge_image_demo.py) also need to add --palette arguments.

File: mmrotate/core/visualization/image.py (hunk 1 above)
Comment (liuyanyi) [reply mmrotate/core/visualization/image.py]: fixed

**Comment by jbwang1997** [comment]
Other datasets can also add the palette.

**Comment by jbwang1997** [comment]
Have you checked the compatibility of mmdet 2.19.0. The palette function is first released in 2.21.0

**Comment by liuyanyi** [comment]
> Have you checked the compatibility of mmdet 2.19.0. The palette function is first released in 2.21.0

This feature need mmdet>=2.22.0.

palette need mmdet>=2.21.0

`draw_labels` and `draw_masks` used in `mmrotate.visualization.image` need mmdet>=2.22.0

**Review by jbwang1997 — APPROVED** [review]
LGTM

**Review by zytx121 — APPROVED** [review]
LGTM

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**